python lint errors

```
Linting python files...
************* Module deploy
hack/dev/kind/gitea/deploy.py:71:20: W3101: Missing timeout argument for method 'requests.get' can cause your program to hang indefinitely (missing-timeout)
hack/dev/kind/gitea/deploy.py:103:15: W3101: Missing timeout argument for method 'requests.post' can cause your program to hang indefinitely (missing-timeout)
hack/dev/kind/gitea/deploy.py:113:15: W3101: Missing timeout argument for method 'requests.post' can cause your program to hang indefinitely (missing-timeout)
hack/dev/kind/gitea/deploy.py:127:15: W3101: Missing timeout argument for method 'requests.post' can cause your program to hang indefinitely (missing-timeout)
hack/dev/kind/gitea/deploy.py:137:8: W3101: Missing timeout argument for method 'requests.delete' can cause your program to hang indefinitely (missing-timeout)
hack/dev/kind/gitea/deploy.py:144:15: W3101: Missing timeout argument for method 'requests.post' can cause your program to hang indefinitely (missing-timeout)
************* Module dev.ghapp_token
hack/dev/ghapp_token.py:85:15: W3101: Missing timeout argument for method 'requests.request' can cause your program to hang indefinitely (missing-timeout)

------------------------------------------------------------------
Your code has been rated at 9.75/10 (previous run: 9.75/10, +0.00)

```

--- a/hack/dev/ghapp_token.py
+++ b/hack/dev/ghapp_token.py
@@ -82,7 +82,9 @@ class GitHub:
             headers.update({"Authorization": "Bearer " + self.token})
         if not url.startswith("http"):
             url = f"{self.github_api_url}{url}"
-        return requests.request(method, url, headers=headers, data=json.dumps(data))
+        return requests.request(
+            method, url, timeout=300, headers=headers, data=json.dumps(data)
+        )
 
 
 def get_private_key(ns):

--- a/hack/dev/kind/gitea/deploy.py
+++ b/hack/dev/kind/gitea/deploy.py
@@ -71,6 +71,7 @@ class ProvisionGitea:
                 r = requests.get(
                     f"{self.gitea_url}/api/v1/version",
                     verify=not OPENSHIFT_ROUTE_FORCE_HTTP,
+                    timeout=300,
                 )
                 if r.status_code == 200:
                     # wait a bit more that it finishes
@@ -105,6 +106,7 @@ class ProvisionGitea:
             data=data_user,
             headers=self.headers,
             verify=not OPENSHIFT_ROUTE_FORCE_HTTP,
+            timeout=300,
         )
         resp.raise_for_status()
 
@@ -113,6 +115,7 @@ class ProvisionGitea:
         resp = requests.post(
             url=f"{self.gitea_url}/api/v1/user/repos",
             headers=self.headers,
+            timeout=300,
             auth=(GITEA_USER, GITEA_PASSWORD),
             verify=not OPENSHIFT_ROUTE_FORCE_HTTP,
             data=jeez,
@@ -129,6 +132,7 @@ class ProvisionGitea:
             headers=self.headers,
             verify=not OPENSHIFT_ROUTE_FORCE_HTTP,
             auth=(GITEA_USER, GITEA_PASSWORD),
+            timeout=300,
             data=jeez,
         )
         resp.raise_for_status()
@@ -138,6 +142,7 @@ class ProvisionGitea:
             url=f"{self.gitea_url}/api/v1/users/{GITEA_USER}/tokens/{self.token_name}",
             headers=self.headers,
             verify=not OPENSHIFT_ROUTE_FORCE_HTTP,
+            timeout=300,
             auth=(GITEA_USER, GITEA_PASSWORD),
         )
         jeez = """{"name": "%s"}""" % (self.token_name)
@@ -146,6 +151,7 @@ class ProvisionGitea:
             headers=self.headers,
             auth=(GITEA_USER, GITEA_PASSWORD),
             verify=not OPENSHIFT_ROUTE_FORCE_HTTP,
+            timeout=300,
             data=jeez,
         )
         resp.raise_for_status()

--- a/pkg/provider/github/github.go
+++ b/pkg/provider/github/github.go
@@ -158,10 +158,14 @@ func makeClient(ctx context.Context, apiURL, token string) (*github.Client, stri
 	return client, providerName, github.String(apiURL)
 }
 
+// checkWebhookSecretValidity check the webhook secret is valid and not
+// ratelimited. we try to check first the header is set (unlimited life token  would
+// not have an expiration) we would anyway get a 401 error when trying to use it
+// but this gives a nice hint to the user into their namespace event of where
+// the issue was
 func (v *Provider) checkWebhookSecretValidity(ctx context.Context, cw clockwork.Clock) error {
 	rl, resp, err := v.Client.RateLimits(ctx)
-	// check if resp.TokenExpiration is after now
-	if resp.TokenExpiration.After(cw.Now()) {
+	if resp.Header.Get("GitHub-Authentication-Token-Expiration") != "" && cw.Now().After(resp.TokenExpiration.Time) {
 		errm := fmt.Sprintf("token has expired at %s", resp.TokenExpiration.Time.Format(time.RFC1123))
 		if err != nil {
 			errm += fmt.Sprintf(" err: %s", err.Error())
@@ -169,9 +173,11 @@ func (v *Provider) checkWebhookSecretValidity(ctx context.Context, cw clockwork.
 		return fmt.Errorf(errm)
 	}
 
+	// some other error happened that is not rate limited related
 	if err != nil {
 		return fmt.Errorf("error using token to access API: %w", err)
 	}
+
 	if rl.SCIM.Remaining == 0 {
 		return fmt.Errorf("token is ratelimited, it will be available again at %s", rl.SCIM.Reset.Time.Format(time.RFC1123))
 	}
@@ -183,7 +189,7 @@ func (v *Provider) SetClient(ctx context.Context, run *params.Run, event *info.E
 	v.providerName = providerName
 	v.Run = run
 
-	// Make sure Client is not already set, so we don't override our fakeclient
+	// check that the Client is not already set, so we don't override our fakeclient
 	// from unittesting.
 	if v.Client == nil {
 		v.Client = client
@@ -192,7 +198,7 @@ func (v *Provider) SetClient(ctx context.Context, run *params.Run, event *info.E
 	v.APIURL = apiURL
 
 	if event.Provider.WebhookSecretFromRepo {
-		// Make sure the webhook secret is still valid
+		// check the webhook secret is valid and not ratelimited
 		if err := v.checkWebhookSecretValidity(ctx, clockwork.NewRealClock()); err != nil {
 			return fmt.Errorf("the webhook secret is not valid: %w", err)
 		}

--- a/pkg/provider/github/github_test.go
+++ b/pkg/provider/github/github_test.go
@@ -591,24 +591,39 @@ func TestGetFiles(t *testing.T) {
 func TestProvider_checkWebhookSecretValidity(t *testing.T) {
 	cw := clockwork.NewFakeClock()
 	tests := []struct {
-		name       string
-		wantSubErr string
-		remaining  int
-		expTime    time.Time
+		name         string
+		wantSubErr   string
+		remaining    int
+		expTime      time.Time
+		expHeaderSet bool
 	}{
 		{
-			name:      "remaining scim calls",
-			remaining: 1,
+			name:         "remaining scim calls",
+			remaining:    1,
+			expHeaderSet: true,
+			expTime:      cw.Now().Add(1 * time.Minute),
 		},
 		{
-			name:       "no remaining scim calls",
-			wantSubErr: "token is ratelimited",
+			name:         "no remaining scim calls",
+			wantSubErr:   "token is ratelimited",
+			remaining:    0,
+			expHeaderSet: true,
+			expTime:      cw.Now().Add(1 * time.Minute),
+		},
+		{
+			name:         "expired",
+			wantSubErr:   "token has expired",
+			expTime:      cw.Now().Add(-1 * time.Minute),
+			expHeaderSet: true,
+		},
+		{
+			name:      "no header mean unlimited",
+			remaining: 5,
+		},
+		{
+			name:       "no header but no remaining scim calls",
 			remaining:  0,
-		},
-		{
-			name:       "expired",
-			wantSubErr: "token has expired",
-			expTime:    cw.Now().Add(1 * time.Minute),
+			wantSubErr: "token is ratelimited",
 		},
 	}
 	for _, tt := range tests {
@@ -628,7 +643,9 @@ func TestProvider_checkWebhookSecretValidity(t *testing.T) {
 				st.Resources = s
 				b, _ := json.Marshal(st)
 				rw.Header().Set("Content-Type", "application/json")
-				rw.Header().Set("GitHub-Authentication-Token-Expiration", tt.expTime.Format("2006-01-02 03:04:05 MST"))
+				if tt.expHeaderSet {
+					rw.Header().Set("GitHub-Authentication-Token-Expiration", tt.expTime.Format("2006-01-02 03:04:05 MST"))
+				}
 				fmt.Fprint(rw, string(b))
 			})
 			defer teardown()

